forum/users/serializers.py

```diff
@@ -82,15 +84,37 @@ def validate_role(self, value):
             raise serializers.ValidationError("Invalid role.")
         return value
 
-    def validate_first_name(self, value):
-        if len(value) > 30:
-            raise serializers.ValidationError("First name must not exceed 30 characters.")
-        return value
 
-    def validate_last_name(self, value):
-        if len(value) > 30:
-            raise serializers.ValidationError("Last name must not exceed 30 characters.")
-        return value
+VALID_TOKEN_ROLES = [Role.STARTUP, Role.INVESTOR]
+
+
+class CustomTokenObtainPairSerializer(TokenObtainPairSerializer):
+    role = serializers.ChoiceField(choices=[(role.value, role.name.lower()) for role in VALID_TOKEN_ROLES])
+    
+    def validate(self, attrs):
+        data = super().validate(attrs)
+
+        request = self.context.get('request')
+        role = request.data.get('role')
+
+        if role is None:
+            raise exceptions.ValidationError({"role": "This field is required."})
+        
+        try:
+            role = int(role)
+            if role not in [role.value for role in VALID_TOKEN_ROLES]:
+                raise ValueError
+        except (ValueError, TypeError):
+            raise exceptions.ValidationError({"role": "Invalid role specified."})
```

Comment:
`ValueError` and `TypeError` are caught generically. This could obscure debugging.
Provide specific error messages and log the invalid inputs.
```
try:
    role = int(role)
    if role not in [role.value for role in VALID_TOKEN_ROLES]:
        raise ValueError(f"Invalid role: {role}")
except ValueError as e:
    raise exceptions.ValidationError({"role": str(e)})
except TypeError:
    raise exceptions.ValidationError({"role": "Role must be an integer."})
```